gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -36,44 +37,41 @@
 public class RepositoryExplorer {
     private static final Logger log = LoggerFactory.getLogger(RepositoryExplorer.class);
 
-    private final String baseUrl;
-    private static final Cache<DependencyGroup, List<Folder>> folderCache = Caffeine.newBuilder()
+    private final Cache<CacheKey, Set<Folder>> folderCache = Caffeine.newBuilder()
             .expireAfterWrite(10, TimeUnit.MINUTES)
             .maximumSize(100)
             .build();
 
-    public RepositoryExplorer(String baseUrl) {
-        this.baseUrl = baseUrl;
-    }
-
-    public final List<Folder> getFolders(DependencyGroup group) {
-        List<Folder> folders = folderCache.get(group, key -> {
-            List<Folder> loadedFolders = loadFolders(key);
+    public final Set<Folder> getFolders(DependencyGroup group, String url) {
```

Comment:
I really think "folder" is not a good name for this. What does it actually represent?

Reply:
It represents one folder in the remote repo, so for example `com` would be a folder and so would `palantir`. so `getFolders` returns a list of all folders based on the current group (which might only be part of a group and not the whole thing) - honestly not sure if it really needs to be a type could probably just use `String`

Reply:
have given a more verbose name of `GroupPartOrPackageName`